### DISCUSSION TITLE
What is the role of Validator nodes ?????

### DISCUSSION THREAD

**Discussion by ghost** [ConsenSys/quorum#1596] [Q&A] [answered]
Hi All,

 I want to use QBFT to build quorum network with 2 participant nodes. So i need to know few things:

 1. I want to understand the difference b/w validators and participant nodes???
 2. Who is going to create validators?? for Exp: As we are going to use 2 participant nodes so how many validators will be required and who will own and control the validator nodes?
 3. What are the minimum no of validators will be required for QBFT??
 4. In case of multi host network setup, What will be the ratio of validators and non-validators nodes per host machine??
 5. Can any participant node have their own validators while adding a new participant node in the running network??

Thanks in advance!!

**Comment by baptiste-b-pegasys** [comment, marked as answer]
1. Validators are nodes participating in the QBFT consensus algorithm to validate blocks. You must have 3 nodes to run, so it is better to have 4 nodes so the network is able to accept one bad validator. The more you have, the more resilient the network will be.

Participant (or member) is a node that represents a private entity for private transactions .(https://consensys.net/docs/goquorum/en/latest/concepts/privacy/private-transaction-lifecycle/?h=participant)
For public txs, they are all the same.

2. it depends on the control you want, if you split the validators by the number of participants, that means 2 validators per party. If you compare with PoW or PoS, the control is shared, if someone gain control, he could provide a new validator and modify the state as he wants.

3. 4 validators, more is better, one bad validator is permitted

https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/consensus-protocols/qbft/#minimum-number-of-validators

https://consensys.net/docs/goquorum/en/latest/concepts/consensus/comparing-poa/

4. you can think the resiliency of the network with the topology you want. If you have two parties with 3 networks for both, you can design with 6 validators, one validator in each net.
The network will run if you have the majority (2f+1). If you have 6 registered validators, you need 4 validators, so 2 validators can go down, but if the two bad validators are in the same party, the other party could gain some control. In this case, if you don't trust the other party, it is better to shutdown the third validator so that the network freeze.

All matter of resiliency, control.

5. Yes, adding a node is subject to vote, so if the controlled is shared, all the parties will vote.

Final note: validator vote on the public state and public transactions, this is why they don't need tessera, but they could be the participant as well.

**Reply by ghost** [reply]
Thanks for reply.

As per your explanation, Validator nodes are only for public tx if we want to do private tx then we will have to add participant nodes with tessera?

And could you please explain again to below :
"you can think the resiliency of the network with the topology you want. If you have two parties with 3 networks for both, you can design with 6 validators, one validator in each net."

**Reply by antonydenyer** [reply]
Private transactions till emit some information (e.g from and to) whilst the payload is not available publicly the key to the payload is. That's what gets validated. Note, there are no protocol level guarantees that each member has the same payload. To have the you need to look at https://consensys.net/docs/goquorum/en/latest/concepts/privacy/privacy-enhancements/

**Comment by baptiste-b-pegasys** [comment]
For private TX, you can take a look that privacy enhancements, that can add more checks on how private transactions are executed. Data inside tessera is critical: you can't share them publicly.

When I say 3 networks, it is like having 3 cloud zones for data/ressource availability. If one shutdowns, your network is still operational.
